framework/src/main/java/org/checkerframework/common/basetype/BaseTypeVisitor.java

```diff
@@ -268,13 +270,24 @@ public class BaseTypeVisitor<Factory extends GenericAnnotatedTypeFactory<?, ?, ?
     /** True if "-AwarnRedundantAnnotations" was passed on the command line */
     private final boolean warnRedundantAnnotations;
 
+    /** True if "-AignoreTargetLocation" was passed on the command line */
+    protected final boolean ignoreTargetLocation;
+
     /** True if "-AcheckEnclosingExpr" was passed on the command line */
     private final boolean checkEnclosingExpr;
 
     /** The tree of the enclosing method that is currently being visited. */
     protected @Nullable MethodTree methodTree = null;
 
     /**
+     * Map from String (canonical name of the qualifier) to its TypeUseLocations declared
+     * in @TargetLocations
+     */
+    protected final HashMap<String, List<TypeUseLocation>> AnnoToTargetLocations = new HashMap<>();
```

Comment:
Use lower-case field name, think of better name.

Reply:
How about `annoDeclUseLocations`?